src/nih_wayfinding/app/scripts/routing/abbreviate-filter.js

```diff
@@ -0,0 +1,105 @@
+
+(function () {
+    'use strict';
+
+    var directions = {
+        north: 'N',
+        northeast: 'NE',
+        east: 'E',
+        southeast: 'SE',
+        south: 'S',
+        southwest: 'SW',
+        west: 'W',
+        northwest: 'NW',
+        n: 'N',
```

Comment:
Why is it necessary to have the abbreviations keyed to themselves here?  Might be cleaner to build an array of the lower-case abbreviation values instead, and check membership on that to see if something is already an abbreviation.